processing/src/main/java/org/apache/druid/segment/UnnestStorageAdapter.java

```diff
@@ -73,20 +67,24 @@ public Sequence<Cursor> makeCursors(
       @Nullable QueryMetrics<?> queryMetrics
   )
   {
-    Filter updatedFilter;
-    if (allowSet != null && !allowSet.isEmpty()) {
-      final InDimFilter allowListFilters;
-      allowListFilters = new InDimFilter(dimensionToUnnest, allowSet);
-      if (filter != null) {
-        updatedFilter = new AndFilter(Arrays.asList(filter, allowListFilters));
-      } else {
-        updatedFilter = allowListFilters;
-      }
+    // the filter on the outer unnested column needs to be recreated on the unnested dimension and sent into
+    // the base cursor
+    // currently testing it  out for in filter and selector filter
+    // TBD: boun
+
+    final Filter forBaseFilter;
+    if (filter == null) {
+      forBaseFilter = filter;
     } else {
-      updatedFilter = filter;
+      // if the filter has the unnested column
+      // do not pass it into the base cursor
+      // if there is a filter as d2 > 1 and unnest-d2 < 10
+      // Calcite would push the filter on d2 into the data source
+      // and only the filter on unnest-d2 < 10 will appear here
+      forBaseFilter = filter.getRequiredColumns().contains(outputColumnName) ? null : filter;
```

Comment:
Nit:

```java
final Filter forBaseFilter = filter == null || filter.getRequiredColumns().contains(outputColumnName) ? null : filter;
```

Reply:
This code seems to assume a single filter. In the example, `unnest-d2 < 10`. But, what if my condition is `unnest-d2 < 10 AND unnest_d2 > 5`. Will this code work? And if only the `unnest-d2` appears here, do we need to check the output column name?

To help understand this, it would be good to give a bit more of the query. What, exactly, is `d2` here?

Reply:
I'll add more comment. d2 is the column to be unnested. What I mean is that all filters on columns that exist in the datasource are pushed on to the data source. Only the filters on the unnested column appears here